lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -3168,6 +3178,16 @@ void VisitNestedScopes(ParseNode* pnodeScopeList, ParseNode* pnodeParent, ByteCo
                 // Push the param scope
                 byteCodeGenerator->PushScope(paramScope);
 
+                /*
+                We need this call to happen *before* visiting the param scope. If we have a split scope,
+                and this function has parameters that default to a lambda which uses the implicit "arguments" inside
+                of eval, we need this function's argumentsSymbol to be set already, so that we can mark it as
+                non-locally referenced inside of PostVisitFunction.
+
+                See Github issue #1209 for more.
+                */
+                AddVarsToScope(pnodeScope->sxFnc.pnodeVars, byteCodeGenerator);
```

Comment:
We can restructure things so that two passes are not necessary, i.e., make the arguments symbol available without searching the body scope vars to find it. Going forward, we're going to create real symbols for this, super, and new.target to staunch the trickle of special-casey bugs involving those constructs, so whatever we invent to allow easy processing of "arguments" in the param scope will work for those other symbols as well.